File: backend/app/schemas/weather.py
from pydantic import BaseModel

class Current(BaseModel):
    temperature: float
    wind_speed: float
    weather_code: int


class Daily(BaseModel):
    date: str
    temperature_min: float
    temperature_max: float
    weather_code: int

class Weather(BaseModel):
    current: Current
    daily: list[Daily]

File: fronted/src/types/weather.ts
export type Weather = {
  current: {
    temperature: number;
    apparent_temperature: number;
    wind_speed: number;
    weather_code: number;
  };

  daily: {
    date: string;
    temperature_min: number;
    temperature_max: number;
    weather_code: number;
  }[];
};